src/detect-filesha256.c

```diff
@@ -79,57 +79,45 @@ static int DetectFileSha256Setup (DetectEngineCtx *de_ctx, Signature *s, const c
 static int SHA256MatchLookupString(ROHashTable *hash, const char *string)
 {
     uint8_t sha256[32];
-    if (ReadHashString(sha256, string, "file", 88, 64) == 1) {
-        void *ptr = ROHashLookup(hash, &sha256, (uint16_t)sizeof(sha256));
-        if (ptr == NULL)
-            return 0;
-        else
-            return 1;
-    }
-    return 0;
+    FAIL_IF_NOT(ReadHashString(sha256, string, "file", 88, 64) == 1);
+    void *ptr = ROHashLookup(hash, &sha256, (uint16_t)sizeof(sha256));
+    FAIL_IF_NULL(ptr);
+    PASS;
```

Comment:
This doesn't have to be converted, as it's a utility function, not a unit test per se. :)

Reply:
Thanks for the note, I saw that it is a utility function but I thought it also needed to be converted, I even did the same thing in this other PR https://github.com/OISF/suricata/pull/8052. I will resend both